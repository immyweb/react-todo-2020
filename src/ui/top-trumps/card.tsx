import * as React from 'react';

import { TTCard } from './styles';

import { ITheme } from '../../types/theme';
import { ICard } from './types';

interface IProps {
  theme: ITheme;
  className?: string;
  card: ICard;
}

export const Card: React.SFC<IProps> = ({ card: { imageMobile, imageDesktop } }) => {
  return <TTCard className={'tt-card'} imageMobile={imageMobile} imageDesktop={imageDesktop} />;
};
