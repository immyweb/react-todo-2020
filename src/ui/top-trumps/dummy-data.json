{
	"config": {
		"theme": {
			"primary": "#3695a6",
			"secondary": "#266773"
		},
		"targetClubId": "ajsept2018"
	},
	"content": {
		"introScreen": {
			"imageMobile":
				"https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x2.jpg",
			"imageDesktop":
				"https://www.thesun.co.uk/wp-content/uploads/2018/09/Intro@2x.jpg",
			"playButtonText": "Play now"
		},
		"voteScreen": {
			"headline": "It's time to decide...",
			"subHeadline": "Who should AJ fight next?",
			"bottomCopy": "Cast your vote and see what others are thinking too.",
			"imageMobile":
				"https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_mob@2x.jpg",
			"imageDesktop":
				"https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_desktop@2x.jpg"
		},
		"resultsScreen": {
			"headline": "The results",
			"subHeadline": "Who should AJ fight next?",
			"linkText": "Boxing news",
			"linkUrl": "https://www.thesun.co.uk/sport/boxing/",
			"imageMobile":
				"https://www.thesun.co.uk/wp-content/uploads/2018/09/results_mob@2x.jpg",
			"imageDesktop":
				"https://www.thesun.co.uk/wp-content/uploads/2018/09/results_desktop@2x.jpg"
		},
		"cards": [
			{
				"id": "wilder",
				"fName": "Deontay",
				"sName": "Wilder",
				"imageThumb":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
				"imageMobile":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg",
				"imageDesktop":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg"
			},
			{
				"id": "whyte",
				"fName": "Dillian",
				"sName": "Whyte",
				"imageThumb":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
				"imageMobile":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg",
				"imageDesktop":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg"
			},
			{
				"id": "fury",
				"fName": "Tyson",
				"sName": "Fury",
				"imageThumb":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
				"imageMobile":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
				"imageDesktop":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg"
			},
			{
				"id": "miller",
				"fName": "Jarrell",
				"sName": "Miller",
				"imageThumb":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-miller@2x.jpg",
				"imageMobile":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/miller-mob@2x.jpg",
				"imageDesktop":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/MILLER@2x.jpg"
			},
			{
				"id": "pulev",
				"fName": "Kubrat",
				"sName": "Pulev",
				"imageThumb":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-pulev@2x.jpg",
				"imageMobile":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/pulev-mob@2x.jpg",
				"imageDesktop":
					"https://www.thesun.co.uk/wp-content/uploads/2018/08/PULEV@2x.jpg"
			}
		]
	}
}
