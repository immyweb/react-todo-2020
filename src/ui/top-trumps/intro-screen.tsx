import * as React from 'react';

import { TTIntroInner, TTWrapper, TTButton, TTIntroButtonHolder } from './styles';

import { IIntroScreen } from './types';
import { ITheme } from '../../types/theme';

interface IProps extends IIntroScreen {
  className?: string;
  theme: ITheme;
  onStartGame(): void;
}

export const IntroScreen: React.SFC<IProps> = ({
  playButtonText,
  theme,
  onStartGame,
  imageMobile,
  imageDesktop,
}) => {
  return (
    <TTIntroInner
      className={'tt-intro-inner'}
      imageMobile={imageMobile}
      imageDesktop={imageDesktop}
    >
      <TTWrapper>
        <TTIntroButtonHolder className={'tt-intro-button-holder'}>
          <TTButton
            className="tt-intro-button"
            backgroundColour={theme.primary}
            hoverBackgroundColour={theme.secondary}
            onClick={onStartGame}
          >
            {playButtonText}
          </TTButton>
        </TTIntroButtonHolder>
      </TTWrapper>
    </TTIntroInner>
  );
};
