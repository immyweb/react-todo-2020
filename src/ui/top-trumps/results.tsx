import * as React from 'react';

import { IResultsScreen, IVoteCount, ICard } from './types';
import { ITheme } from '../../types/theme';
import { TTDefaultInner, TTResultsButtonLink, TTWrapper, TTResultsChartContainer } from './styles';
import { ChartHorizontalBar } from './horizontal-bar-chart';

interface IProps extends IResultsScreen {
  className: string;
  theme: ITheme;
}

export const processData = (cards: ICard[]): IVoteCount[] => {
  let totalVoteCount = 0;
  return cards.map(
    ({
      fName,
      sName,
      imageThumb,
      voteCount: { voteCount, noVoteCount } = { voteCount: 0, noVoteCount: 0 },
    }) => {
      totalVoteCount += voteCount;
      const datum = {
        imageThumb,
        voteCount,
        noVoteCount,
        player: `${fName} ${sName}`,
        get votePercentage(): number {
          return totalVoteCount ? Math.round((voteCount / totalVoteCount) * 100) : 0;
        },
      };
      return datum;
    },
  );
};

export const Results: React.SFC<IProps> = ({
  theme,
  cards,
  linkText,
  linkClick,
  imageDesktop,
  imageMobile,
}) => {
  const graphData = processData(cards);

  return (
    <TTDefaultInner imageMobile={imageMobile} imageDesktop={imageDesktop}>
      <TTWrapper>
        <TTResultsChartContainer>
          <ChartHorizontalBar data={graphData} theme={theme} />
        </TTResultsChartContainer>
        <TTResultsButtonLink
          onClick={linkClick}
          colour={theme.primary}
          hoverBackgroundColour={theme.secondary}
        >
          {linkText}
        </TTResultsButtonLink>
      </TTWrapper>
    </TTDefaultInner>
  );
};
