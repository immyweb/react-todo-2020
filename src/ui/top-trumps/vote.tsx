import * as React from 'react';

import { IVoteScreen } from './types';
import { ITheme } from '../../types/theme';
import {
  TTDefaultInner,
  TTVoteButtonContainer,
  TTVoteButton,
  TTVoteButtonThumb,
  TTVoteButtonThumbContainer,
  TTVoteButtonText,
  TTWrapper,
} from './styles';

interface IProps extends IVoteScreen {
  className: string;
  theme: ITheme;
  clickVote: (cardIds: string[]) => void;
}

export const Vote: React.SFC<IProps> = ({ theme, cards, clickVote, imageMobile, imageDesktop }) => {
  return (
    <TTDefaultInner imageMobile={imageMobile} imageDesktop={imageDesktop}>
      <TTWrapper>
        <TTVoteButtonContainer>
          {cards.map(({ id, fName, sName, imageThumb }) => (
            <TTVoteButton
              key={sName}
              onClick={() => clickVote([id])}
              colour={theme.primary}
              hoverBackgroundColour={theme.secondary}
            >
              <TTVoteButtonThumbContainer>
                <TTVoteButtonThumb src={imageThumb} />
              </TTVoteButtonThumbContainer>
              <TTVoteButtonText colour={theme.secondary}>{`${fName} ${sName}`}</TTVoteButtonText>
            </TTVoteButton>
          ))}
        </TTVoteButtonContainer>
      </TTWrapper>
    </TTDefaultInner>
  );
};
