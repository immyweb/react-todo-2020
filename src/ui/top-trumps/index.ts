import { TopTrumps } from './top-trumps';

export default TopTrumps;
