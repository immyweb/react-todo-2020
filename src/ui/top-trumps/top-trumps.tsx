import * as React from 'react';

import { transferTinderAPI } from '../../api-services';
import { IntroScreen } from './intro-screen';
import { Card } from './card';
import { Vote } from './vote';
import { ProgressBar } from './progress-bar';
import { Results } from './results';
import Error from '../error';

import { TTContainer, TTGame } from './styles';

import { IComponentProps } from '../../types/tracking';
import { ITheme } from '../../types/theme';
import { IIntroScreen, IVoteScreen, IResultsScreen, IVoteCount, ICard } from './types';

interface IProps extends IComponentProps {
  theme: ITheme;
  targetClubId: string;
  cards: ICard[];
  introScreen: IIntroScreen;
  voteScreen: IVoteScreen;
  resultsScreen: IResultsScreen;
}

interface IState {
  currentCardIndex: number;
  cards: ICard[];
  showIntro: boolean;
  showCards: boolean;
  showVote: boolean;
  showResults: boolean;
  hasErrored: boolean;
  voteCount?: IVoteCount;
  currentCard?: ICard;
}

export class TopTrumps extends React.Component<IProps, IState> {
  cachedWarmState = {};

  initialState = {
    currentCardIndex: 0,
    cards: [],
    showIntro: true,
    showCards: false,
    showVote: false,
    showResults: false,
    hasErrored: false,
  };

  constructor(props: IProps) {
    super(props);

    this.state = {
      ...this.initialState,
      cards: props.cards,
    };
  }

  async componentDidMount() {
    const { targetClubId, cards } = this.props;
    const cardIds = cards.map((card: ICard) => card.id);
    try {
      const cardVotes = await transferTinderAPI.getPlayerVotes(targetClubId, cardIds);

      const cardsWithVotes = this.mergeVotes(cardVotes, cards) as ICard[];
      this.setState({ cards: cardsWithVotes });
    } catch (e) {
      this.setState({ hasErrored: true });
    }
  }

  mergeVotes = (results: IVoteCount[], cards: ICard[]) => {
    return cards.map((card) => {
      const matchingResult = results.find((result) => {
        return card.id === result.player;
      });
      return matchingResult ? { ...card, voteCount: matchingResult } : { ...card };
    });
  };

  onStartGame = () => {
    this.setState({
      showIntro: false,
      showCards: true,
    });
  };

  clickLink = () => {
    window.location.assign(this.props.resultsScreen.linkUrl);
  };

  clickVote = async (votesYes: string[]) => {
    const { targetClubId } = this.props;
    this.setState({
      showVote: false,
      showResults: true,
    });

    await transferTinderAPI.addPlayerVotes(targetClubId, votesYes, []);
  };

  clickLeft = () => {
    this.setState({
      currentCardIndex: Math.max(0, this.state.currentCardIndex - 1),
    });
  };

  clickRight = () => {
    const { currentCardIndex, cards } = this.state;
    if (currentCardIndex === cards.length - 1) {
      this.setState({
        showCards: false,
        showVote: true,
      });
    } else {
      this.setState({
        currentCardIndex: Math.min(cards.length, currentCardIndex + 1),
      });
    }
  };

  render() {
    const { introScreen, resultsScreen, voteScreen, theme } = this.props;
    const { showIntro, showCards, showVote, showResults, cards, currentCardIndex, hasErrored } = this.state;

    return hasErrored ? (
      <Error />
    ) : (
      <TTContainer className="ttt-container">
        {showIntro && (
          <IntroScreen className="ttt-intro-screen" {...introScreen} onStartGame={this.onStartGame} theme={theme} />
        )}
        {showCards && (
          <TTGame className="tt-game">
            <Card className="tt-card" card={cards[currentCardIndex]} theme={theme} />
            <ProgressBar
              className="tt-progress_bar"
              cards={cards}
              currentCardIndex={currentCardIndex}
              theme={theme}
              onClickLeft={this.clickLeft}
              onClickRight={this.clickRight}
            />
          </TTGame>
        )}
        {showVote && (
          <Vote className="tt-vote" {...voteScreen} cards={cards} theme={theme} clickVote={this.clickVote} />
        )}
        {showResults && (
          <Results className="tt-results" {...resultsScreen} cards={cards} theme={theme} linkClick={this.clickLink} />
        )}
      </TTContainer>
    );
  }
}
