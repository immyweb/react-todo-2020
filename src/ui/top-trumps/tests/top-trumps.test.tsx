import * as React from 'react';
import { shallow } from 'enzyme';

import { TopTrumps } from '../top-trumps';
import { IntroScreen } from '../intro-screen';
import { Vote } from '../vote';
import { Results } from '../results';
import { TTGame } from '../styles';

import { podTracker } from '../../../utils/pod-tracker-utils';

jest.mock('../../../api-services', () => ({
  transferTinderAPI: {
    addPlayerVotes: jest.fn(),
  },
}));

jest.mock('../../../utils/pod-tracker-utils', () => ({
  podTracker: {
    click: jest.fn(),
  },
}));

const theme = {
  primary: '#ffffff',
  secondary: '#000000',
};

const introScreen = {
  imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg',
  imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg',
  playButtonText: 'Play now',
};

const resultsScreen = {
  headline: 'The results',
  subHeadline: 'Who should AJ fight next?',
  linkText: 'Boxing news',
  linkUrl: 'https://www.thesun.co.uk/sport/boxing/',
  imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/09/results_mob@2x.jpg',
  imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/09/results_desktop@2x.jpg',
};

const voteScreen = {
  headline: "It's time to decide...",
  subHeadline: 'Who should AJ fight next?',
  bottomCopy: 'Cast your vote and see what others are thinking too.',
  imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_mob@2x.jpg',
  imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_desktop@2x.jpg',
};

const cards = [
  {
    id: 'wilder',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg',
    fName: 'Deontay',
    sName: 'Wilder',
  },
  {
    id: 'whyte',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg',
    fName: 'Dillian',
    sName: 'Whyte',
  },
  {
    id: 'fury',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg',
    fName: 'Tyson',
    sName: 'Fury',
  },
];

function shallowRender(props?: any) {
  return shallow(
    <TopTrumps
      id={'card'}
      theme={theme}
      targetClubId={'ajsept2018'}
      introScreen={introScreen}
      resultsScreen={resultsScreen}
      voteScreen={voteScreen}
      cards={cards}
      {...props}
    />,
  );
}

describe('<TopTrumps />', () => {
  it('renders correctly when errored', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when not errored', () => {
    const wrapper = shallowRender();
    wrapper.setState({
      hasErrored: false,
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('sets correct state values on launch', () => {
    const wrapper = shallowRender();
    expect(wrapper.state('showIntro')).toEqual(true);
    expect(wrapper.state('showCards')).toEqual(false);
    expect(wrapper.state('showVote')).toEqual(false);
    expect(wrapper.state('showResults')).toEqual(false);
  });

  it('onStartGame hides <IntroScreen />, shows <TTGame /> and tracks event', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as TopTrumps;
    wrapper.setState({
      hasErrored: false,
    });
    instance.onStartGame();
    expect(wrapper.state('showIntro')).toEqual(false);
    expect(wrapper.state('showCards')).toEqual(true);
    expect(wrapper.state('showVote')).toEqual(false);
    expect(wrapper.state('showResults')).toEqual(false);
    wrapper.update();
    const introScreen = wrapper.find(IntroScreen);
    expect(introScreen.length).toEqual(0);
    const gameScreen = wrapper.find(TTGame);
    expect(gameScreen.length).toEqual(1);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Top Trumps game started',
      id: 'card',
    });
  });

  it('clickLink function goes to supplied exit url', () => {
    window.location.assign = jest.fn();
    const wrapper = shallowRender();
    const instance = wrapper.instance() as TopTrumps;
    instance.clickLink();
    expect(window.location.assign).toBeCalledWith('https://www.thesun.co.uk/sport/boxing/');
  });

  it('clickLeft function sets correct currentCardIndex index and tracks event', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as TopTrumps;
    wrapper.setState({
      hasErrored: false,
    });
    instance.clickLeft();
    expect(wrapper.state('currentCardIndex')).toEqual(0);
    wrapper.setState({
      currentCardIndex: 3,
    });
    instance.clickLeft();
    expect(wrapper.state('currentCardIndex')).toEqual(2);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Top Trumps click back',
      id: 'card',
    });
  });

  it('clickRight function sets correct currentCardIndex index, correct state, and tracks event', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as TopTrumps;
    wrapper.setState({
      hasErrored: false,
    });
    instance.clickRight();
    expect(wrapper.state('currentCardIndex')).toEqual(1);
    instance.clickRight();
    expect(wrapper.state('currentCardIndex')).toEqual(2);
    instance.clickRight();
    expect(wrapper.state('showCards')).toEqual(false);
    expect(wrapper.state('showVote')).toEqual(true);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Top Trumps click forward',
      id: 'card',
    });
  });

  it('mergeVotes function adds correct vote count to player', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as TopTrumps;
    const results = [
      {
        player: 'wilder',
        voteCount: 5,
        noVoteCount: 5,
      },
      {
        player: 'whyte',
        voteCount: 10,
        noVoteCount: 0,
      },
      {
        player: 'fury',
        voteCount: 0,
        noVoteCount: 10,
      },
    ];
    const result = instance.mergeVotes(results, cards);
    expect(result).toEqual([
      {
        id: 'wilder',
        imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg',
        imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg',
        imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg',
        fName: 'Deontay',
        sName: 'Wilder',
        voteCount: { player: 'wilder', voteCount: 5, noVoteCount: 5 },
      },
      {
        id: 'whyte',
        imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg',
        imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg',
        imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg',
        fName: 'Dillian',
        sName: 'Whyte',
        voteCount: { player: 'whyte', voteCount: 10, noVoteCount: 0 },
      },
      {
        id: 'fury',
        imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg',
        imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg',
        imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg',
        fName: 'Tyson',
        sName: 'Fury',
        voteCount: { player: 'fury', voteCount: 0, noVoteCount: 10 },
      },
    ]);
  });

  it('clickVote update the state, track an event, and send the vote data to the api', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as TopTrumps;
    wrapper.setState({
      hasErrored: false,
      showIntro: false,
      showCards: false,
      showVote: true,
      showResults: false,
    });
    instance.clickVote(['fury']);
    expect(wrapper.state('showIntro')).toEqual(false);
    expect(wrapper.state('showCards')).toEqual(false);
    expect(wrapper.state('showVote')).toEqual(false);
    expect(wrapper.state('showResults')).toEqual(true);
    wrapper.update();
    const voteScreen = wrapper.find(Vote);
    expect(voteScreen.length).toEqual(0);
    const resultsScreen = wrapper.find(Results);
    expect(resultsScreen.length).toEqual(1);

    expect(podTracker.click).toHaveBeenCalledWith({
      description: 'Generic Top Trumps click vote',
      id: 'card',
    });
  });
});
