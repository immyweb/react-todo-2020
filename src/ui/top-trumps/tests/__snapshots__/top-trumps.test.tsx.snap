// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopTrumps /> renders correctly when errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopTrumps
    cards={
      Array [
        Object {
          "fName": "Deontay",
          "id": "wilder",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
          "sName": "Wilder",
        },
        Object {
          "fName": "Dillian",
          "id": "whyte",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
          "sName": "Whyte",
        },
        Object {
          "fName": "Tyson",
          "id": "fury",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
          "sName": "Fury",
        },
      ]
    }
    id="card"
    introScreen={
      Object {
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg",
        "playButtonText": "Play now",
      }
    }
    resultsScreen={
      Object {
        "headline": "The results",
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/results_desktop@2x.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/results_mob@2x.jpg",
        "linkText": "Boxing news",
        "linkUrl": "https://www.thesun.co.uk/sport/boxing/",
        "subHeadline": "Who should AJ fight next?",
      }
    }
    targetClubId="ajsept2018"
    theme={
      Object {
        "primary": "#ffffff",
        "secondary": "#000000",
      }
    }
    voteScreen={
      Object {
        "bottomCopy": "Cast your vote and see what others are thinking too.",
        "headline": "It's time to decide...",
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_desktop@2x.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_mob@2x.jpg",
        "subHeadline": "Who should AJ fight next?",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backgroundColour": "#ffffff",
      "theme": Object {
        "primary": "#eb1701",
        "secondary": "#a5171c",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backgroundColour": "#ffffff",
        "theme": Object {
          "primary": "#eb1701",
          "secondary": "#a5171c",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<TopTrumps /> renders correctly when not errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopTrumps
    cards={
      Array [
        Object {
          "fName": "Deontay",
          "id": "wilder",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
          "sName": "Wilder",
        },
        Object {
          "fName": "Dillian",
          "id": "whyte",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
          "sName": "Whyte",
        },
        Object {
          "fName": "Tyson",
          "id": "fury",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
          "sName": "Fury",
        },
      ]
    }
    id="card"
    introScreen={
      Object {
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg",
        "playButtonText": "Play now",
      }
    }
    resultsScreen={
      Object {
        "headline": "The results",
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/results_desktop@2x.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/results_mob@2x.jpg",
        "linkText": "Boxing news",
        "linkUrl": "https://www.thesun.co.uk/sport/boxing/",
        "subHeadline": "Who should AJ fight next?",
      }
    }
    targetClubId="ajsept2018"
    theme={
      Object {
        "primary": "#ffffff",
        "secondary": "#000000",
      }
    }
    voteScreen={
      Object {
        "bottomCopy": "Cast your vote and see what others are thinking too.",
        "headline": "It's time to decide...",
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_desktop@2x.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/vote_mob@2x.jpg",
        "subHeadline": "Who should AJ fight next?",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          className="ttt-intro-screen"
          imageDesktop="https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg"
          imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg"
          onStartGame={[Function]}
          playButtonText="Play now"
          theme={
            Object {
              "primary": "#ffffff",
              "secondary": "#000000",
            }
          }
        />,
        false,
        false,
        false,
      ],
      "className": "ttt-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "ttt-intro-screen",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg",
          "onStartGame": [Function],
          "playButtonText": "Play now",
          "theme": Object {
            "primary": "#ffffff",
            "secondary": "#000000",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            className="ttt-intro-screen"
            imageDesktop="https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg"
            imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg"
            onStartGame={[Function]}
            playButtonText="Play now"
            theme={
              Object {
                "primary": "#ffffff",
                "secondary": "#000000",
              }
            }
          />,
          false,
          false,
          false,
        ],
        "className": "ttt-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "ttt-intro-screen",
            "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_desktop@2x1.jpg",
            "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/09/intro_mob@2x1.jpg",
            "onStartGame": [Function],
            "playButtonText": "Play now",
            "theme": Object {
              "primary": "#ffffff",
              "secondary": "#000000",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
